test/distributed/_shard/test_partial_tensor.py

```diff
@@ -39,30 +39,33 @@ class TestPartialTensorReshard(ShardedTensorTestBase):
     def _run_partial_tensor_n_reshard(
         self, reshard_spec, input_size, world_size, reduce_op, dtype=torch.float
```

Comment:
`resharding_spec` does come with different orders of placements and also there are cases when we need to do padding. So both cases are covered here. @pritamdamania87